src/extra/gfx/camera.cpp

```diff
@@ -26,91 +26,141 @@ struct FreeCameraShard : public gfx::BaseConsumer {
   PARAM_IMPL(FreeCameraShard, PARAM_IMPL_FOR(_flySpeed), PARAM_IMPL_FOR(_scrollSpeed), PARAM_IMPL_FOR(_panSpeed),
              PARAM_IMPL_FOR(_lookSpeed));
 
+  // Mouse/Right stick rotation input and mouse X/Y movement (drag)
+  struct PointerInputState {
+    float2 position{};
+    float2 prevPosition{};
+    bool primaryButton{};
+    bool secondaryButton{};
+    bool tertiaryButton{};
+  };
+
+  // Input state for a button axis
+  struct AxisInputState {
+    bool pos{};
+    bool neg{};
+
+    float getValue() const {
+      if (pos)
+        return 1.0f;
+      if (neg)
+        return -1.0f;
+      return 0.0f;
+    }
+  };
+
+  // Tracked cursor & button states
   struct InputState {
-    struct Keys {
-      bool left{};
-      bool right{};
-      bool forward{};
-      bool backward{};
-    } keys;
-    struct Look {
-      float2 cursorPosition{};
-      float2 prevCursorPosition{};
-      bool primary{};
-      bool secondary{};
-      bool tertiary{};
-    } look;
-    float zoom{};
-  } _input;
+    AxisInputState keyboardXAxis;
+    AxisInputState keyboardZAxis;
+    PointerInputState pointer;
+    float mouseWheel{};
+    float deltaTime;
+  } _inputState;
+
+  // Inputs for camera movement
+  struct CameraInputs {
+    // XYZ camera translation (time scaled)
+    float3 velocity;
+    // XYZ camera translation (absolute)
+    float3 movement;
```

Comment:
I think velocity and translation describe the variables more closely (assuming translation = velocity * t)